cf-pubsub/src/main/java/org/eclipse/californium/pubsub/Converter.java

```diff
@@ -0,0 +1,264 @@
+    /*helper functions for  CoapResponse */
+
+    /**
+     * creates a Set of WebLinks from the ResponseText of a given CoapResponse
+     *
+     * @param response CoapResponse
+     * @return a Set of WebLink
+     */
+    public static Set<WebLink> getWebLinks(CoapResponse response){
+        return LinkFormat.parse(response.getResponseText());
+    }
+
+    /**
+     * renders a WebLink for each available topic in the ResponseText of a given CoapResponse
+     * This would include the node topics that have child/leaf topics in the returned Set
+     *
+     * @param response CoapResponse
+     * @return a Set of WebLink
+     */
+    public static Set<WebLink> getAllWebLinks(CoapResponse response) {
+        return extractAllWebLinks(getWebLinks(response));
+    }
+
+    /*helper functions for Set<WebLink>*/
+
+    /**
+     * renders an array out of a given Set of WebLinks
+     * useful if you want to access a specific WebLink without casting to an ArrayList
+     *
+     * @param webLinks Set<WebLink>
+     * @return an array of WebLinks
+     */
+    public static WebLink[] getArray(Set<WebLink> webLinks) {
+        return webLinks.toArray(new WebLink[0]);
+    }
+
+
+    /**
+     * renders a WebLink for each available topic in the uris of the WebLinks in the given Set
+     * This would include the node topics that have child/leaf topics in the returned Set
+     *
+     * @param webLinks Set<WebLink>
+     * @return a Set of WebLink
+     */
+    public static Set<WebLink> extractAllWebLinks(Set<WebLink> webLinks) {
+        Set<WebLink> topics = new ConcurrentSkipListSet<WebLink>();
+        WebLink p;
+        for(WebLink w: webLinks) {
+            p = w;
+            while (!cleanUri(p.getURI()).equals("ps")) {
+                topics.add(p);
+                p = getParent(p);
+            }
+        }
+        return topics;
+    }
+
+    /*helper functions for WebLink*/
+
+    /**
+     * renders a WebLink using the given uri and sets its Content type
+     *
+     * @param uri String
+     * @param ct int
+     * @return a WebLink
+     */
+    public static WebLink makeWebLink(String uri, int ct) {
+        WebLink webLink = new WebLink('/' + cleanUri(uri));
+        setCT(webLink, ct);
+        return webLink;
+    }
+
+    /**
+     * gets a uri usable with the {@link PubSub} functions
+     * it uses {@link #cleanUri(String)} to trim slashes from the
+     * beginning and ending of the WebLink's uri
+     *
+     * @param webLink WebLink
+     * @return a String uri
+     */
+    public static String getUri(WebLink webLink) {
+        return cleanUri(webLink.getURI());
+    }
+
+    /**
+     * returns an int of the ContentType of the given WebLink
+     * it parses the ContentType returned to ensure it does not contain extra letters
+     * due to issues faced with the testing broker we used
+     *
+     * @param webLink WebLink
+     * @return an int ContentType
+     */
+    public static int getCT(WebLink webLink) {
```

Comment:
Usually we try not to use abbreviations. Please consider to rename it to `getContentType`.

Reply:
Also for `setCT` an `getCTString`.